akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -422,6 +425,10 @@ class ClusterSingletonManager(
     removed = removed filter { case (_, deadline) ⇒ deadline.hasTimeLeft }
   }
 
+  // for CoordinatedShutdown
+  val memberExitingProgress = Promise[Done]()
+  CoordinatedShutdown(context.system).addTask("cluster-pre-exiting")(() ⇒ memberExitingProgress.future)
```

Comment:
Is there a case where we'd complete the `memberExitingProgress ` with a failure, like a timeout in this process?